src/Microsoft.TestPlatform.ObjectModel/TestServiceLocator.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+// We don't want this in our shipped code. Build only for debug until I am able to remove it.
+#if DEBUG
+
+#if !NETSTANDARD1_0
+using System;
+#endif
+using System.Collections.Generic;
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+#pragma warning disable RS0037 // Enable tracking of nullability of reference types in the declared API
+
+public static class TestServiceLocator
+{
+    public static Dictionary<string, object> Instances { get; } = new Dictionary<string, object>();
+    public static List<Resolve> Resolves { get; } = new();
+
+    public static void Register<TRegistration>(string name, TRegistration instance) where TRegistration : notnull
+    {
+        Instances.Add(name, instance);
```

Comment:
[styling] Any preference between body and `=>` for one line methods/properties? I could turn on some rules.

Reply:
No strong preference, I would not impose more rules, until we can get the current rule set to work 100% all the time.